tools/Jenkins/dependency_spotter.py

```diff
@@ -0,0 +1,143 @@
+import argparse
+import re
+import urllib.request
+
+
+def dependency_spotter(os_name: str, first_build: int, second_build: int, pipeline: str, log_file: str):
+    if second_build < first_build:
+        yes_no = input("It looks like the second build is older than the first build, do you want to swap the order? (y/n)")
+        if yes_no.lower() == "y":
+            first_build, second_build = second_build, first_build
```

Comment:
Might need some feedback? just thinking if someone types 'yes' they should know it didn't do what they expected. Or maybe it should just also accept 'yes'.